components/net/bluetooth_thread.rs

```diff
@@ -359,28 +362,30 @@ impl BluetoothManager {
             send_error!(sender, "No device found");
         }
 
-        //TODO select the proper device
-        let device = &devices[0];
-
-        let message = BluetoothObjectMsg::BluetoothDevice {
-            id: device.get_address().unwrap_or("".to_owned()),
-            name: device.get_name().ok(),
-            device_class: device.get_class().ok(),
-            vendor_id_source: device.get_vendor_id_source().ok(),
-            vendor_id: device.get_vendor_id().ok(),
-            product_id: device.get_product_id().ok(),
-            product_version: device.get_device_id().ok(),
-            appearance: device.get_appearance().ok(),
-            tx_power: match device.get_tx_power() {
-                Ok(p) => Some(p as i8),
-                Err(_) => None,
-            },
-            rssi: match device.get_rssi() {
-                Ok(p) => Some(p as i8),
-                Err(_) => None,
-            }
-        };
-        sender.send(message).unwrap();
+        match devices.into_iter().find(|ref d| matches_filters(d, options.get_filters())) {
+            Some(device) => {
+                let message = BluetoothObjectMsg::BluetoothDevice {
+                    id: device.get_address().unwrap_or("".to_owned()),
+                    name: device.get_name().ok(),
+                    device_class: device.get_class().ok(),
+                    vendor_id_source: device.get_vendor_id_source().ok(),
+                    vendor_id: device.get_vendor_id().ok(),
+                    product_id: device.get_product_id().ok(),
+                    product_version: device.get_device_id().ok(),
+                    appearance: device.get_appearance().ok(),
+                    tx_power: match device.get_tx_power() {
+                        Ok(p) => Some(p as i8),
+                        Err(_) => None,
+                    },
```

Comment:
Or we can keep this code and use Err(error) insted of Err(_) and send back the error @dati91?